source/MaterialXGenOsl/OslShaderGenerator.cpp

```diff
@@ -181,6 +182,17 @@ OslShaderGenerator::OslShaderGenerator() :
 
     // <!-- <surface> -->
     registerImplementation("IM_surface_" + OslShaderGenerator::TARGET, SurfaceNodeOsl::create);
+
+    // <!-- <image> -->
+    registerImplementation("IM_image_float_" + OslShaderGenerator::TARGET, SourceCodeNode::create);
+    registerImplementation("IM_image_color3_" + OslShaderGenerator::TARGET, SourceCodeNode::create);
+    registerImplementation("IM_image_color4_" + OslShaderGenerator::TARGET, SourceCodeNode::create);
+    registerImplementation("IM_image_vector2_" + OslShaderGenerator::TARGET, SourceCodeNode::create);
+    registerImplementation("IM_image_vector3_" + OslShaderGenerator::TARGET, SourceCodeNode::create);
+    registerImplementation("IM_image_vector4_" + OslShaderGenerator::TARGET, SourceCodeNode::create);
```

Comment:
This custom registratin should not be needed now. Since you solved the extra arguments using a token replacement, and the value assignment using the TypeSyntax classes, there is no custom codegen needed for these nodes.
(The `SourceCodeNode` is the default fallback implementation used, so this extra registration is redundant)

Reply:
Will fix.